scripts/lib/mkcloud-common.sh

```diff
@@ -1,5 +1,16 @@
 # This file shares common code between mkcloud-${mkclouddriver}.sh files and qa_crowbarsetup.sh
 
+function to_ip
+{
+    safely getent hosts "$1" | head -n1 | awk '{print $1}' | grep .
```

Comment:
This is not a robust approach to ensuring that the exit code of the pipeline reflects the success of the first component in the pipeline (i.e. the `safely getent hosts "$1"`).  It assumes that the first component is successful if and only if it emits something to STDOUT, but there are two risks with that:

1. `getent hosts` might emit to STDOUT and still fail. Even if it doesn't now, it might in future versions.
2. If `safely` or one of the functions it calls, such as `is_onhost` or `complain` changes in the future, it could also emit something to STDOUT.

So this is brittle.  I think it would be better to rely purely on the exit code, which is the correct semantics for checking success.  For this [you can either use `PIPESTATUS` or `pipefail`](http://stackoverflow.com/questions/1221833/pipe-output-and-capture-exit-status-in-bash).